nullaway/src/main/java/com/uber/nullaway/handlers/CompositeHandler.java

```diff
@@ -264,4 +265,20 @@ public MethodInvocationNode onCFGBuildPhase1AfterVisitMethodInvocation(
     }
     return currentNode;
   }
+
+  @Override
+  @Nullable
+  public Integer castToNonNullArgumentPositionsForMethod(
+      NullAway analysis,
+      VisitorState state,
+      Symbol.MethodSymbol methodSymbol,
+      List<? extends ExpressionTree> actualParams,
+      @Nullable Integer previousArgumentPosition) {
+    for (Handler h : handlers) {
+      previousArgumentPosition =
```

Comment:
This still seems a bit strange to me, like we are "accumulating" the position across the handlers.  Could we just get rid of the `previousArgumentPosition` parameter, and just return the first non-null value given by some handler, and `null` otherwise?  One might argue that is error prone, but I don't think it is any more so than what we do here (since a later handler would override a previous hander's answer), and I don't think we'll be writing tons of changing code around this in the future.  Or am I missing something?

Reply:
There is a general pattern with handlers overriding the previous handler's answer, where the handler later in the chain generally takes precedence but can inspect the answer of previous handlers. In this case it doesn't matter, because we have a single handler dealing with these methods, but I would actually find it surprising if we didn't follow the same general pattern of overriding in chain order that we do for _every other non-void return handler callback_...

Reply:
See e.g.:

```
boolean onOverrideMayBeNullExpr(
      NullAway analysis, ExpressionTree expr, VisitorState state, boolean exprMayBeNull);
```

Or:

```
  MethodInvocationNode onCFGBuildPhase1AfterVisitMethodInvocation(
      NullAwayCFGBuilder.NullAwayCFGTranslationPhaseOne phase,
      MethodInvocationTree tree,
      MethodInvocationNode originalNode);
```

p.s. Technically, this is not true of all handler methods, e.g. `onDataflowVisitMethodInvocation` has a custom way of combining `NullnessHint`s based on a lattice merge. But still... it is a common pattern for handlers.

Reply:
I think there are probably ways to streamline the callbacks and standardize the arguments they take, but I don't feel the chain of overrides pattern is a particularly bad idea architecturally. Happy to discuss further, though, separate from this PR :)